lms/static/scripts/frontend_apps/components/dashboard/DashboardActivityFilters.tsx

```diff
@@ -262,6 +288,40 @@ export default function DashboardActivityFilters({
           />
         )}
       />
+      {segments && (
+        <MultiSelect
+          aria-label={`Select ${segments.type}`}
+          containerClasses="!w-auto min-w-[180px]"
+          value={segments.selectedIds}
+          onChange={newSegmentIds => segments.onChange(newSegmentIds)}
+          buttonContent={
+            segments.selectedIds.length === 0 ? (
+              <>All {segments.type}</>
+            ) : segments.selectedIds.length === 1 ? (
+              segments.entries.find(
+                s => s.h_authority_provided_id === segments.selectedIds[0],
+              )?.name ?? `1 ${segments.type.slice(0, -1)}`
```

Comment:
In general I'm not keen on reusing literals that are used as type/enum discriminators to construct UI strings. A future developer might expect that changing the literal would be safe as long as they update references to it, but here it could break in a subtle way. Also we'll eventually have to change the approach entirely when we add translation to the UI.

Reply:
Yep, you are right. This had a smell since the beginning, and it just happens to work now, but it doesn't have to in future.

I'll do a proper mapping between the actual discriminator type and the information displayed in the UI to avoid this issue.

Reply:
I extracted the whole MultiSelect into a local component, and in there I added the logic to resolve a human friendly name in plural and singular based on the segments type.